docs/site/gpu.md

```diff
@@ -65,6 +83,84 @@ CUDA toolkit
   3. [CUDA 10.2](https://developer.nvidia.com/cuda-10.2-download-archive)
   4. [CUDNN 7.x](https://developer.nvidia.com/cudnn)
 
+## Linux
+
+## Linux
+
+One easiest way to install the NVIDIA software is with `apt` on Ubuntu. For other distributions
+refer to the [CUDA install Linux](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html).
+
+Note: All linux distributions may not support this. you might encounter some problems with driver
+installations.
+
+To check the CUDA compatible driver version:
+
+```sh
+
+# linux-modules-nvidia-390-5.0.0-1018-azure
+#                                    -aws, -gcp, -oracle, -generic, or -lowlatency
+
+# Get the latest driver version number, significant digit 5 like 435, 450
+NVIDIA_DRIVER_VERSION=$(sudo apt-cache search 'linux-modules-nvidia-[0-9]+-generic$' | awk '{print $1}' | sort | tail -n 1 | head -n 1 | awk -F"-" '{print $4}')
+
+CUDA_DRIVER_VERSION=$(apt-cache madison nvidia-cuda-toolkit | awk '{print $3}' | sort -r | while read line; do
+   if dpkg --compare-versions $(dpkg-query -f='${Version}\n' -W nvidia-driver-${NVIDIA_DRIVER_VERSION}) ge $line ; then
+       echo "$line"
+       break
+   fi
+done)
+```
+
+Install [CUPTI](http://docs.nvidia.com/cuda/cupti/) which ships with CUDA toolkit for profiling.
+
+```sh
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/extras/CUPTI/lib64
+```
+
+### Install CUDA with apt
+
+The following instructions are for installing CUDA 10.2 on Ubuntu 18.04. These instructions
+might work for other Debian-based distros.
+
+Note: [Secure Boot](https://wiki.ubuntu.com/UEFI/SecureBoot) tends to complication installation.
+These instructions may not address this.
+
+#### Ubuntu 18.04 (CUDA 10.2)
+
+```sh
+
+# Add NVIDIA package repositories
+# 1. Download the Ubuntu 18.04 driver repository
+wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-ubuntu1804.pin
+# 2. Move the repository to preferences
+sudo mv cuda-ubuntu1804.pin /etc/apt/preferences.d/cuda-repository-pin-600
+# 3. Fetch keys
+sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/7fa2af80.pub
+# 4. add repository
+sudo add-apt-repository "deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/ /"
+# 5. Update package lists
+sudo apt-get update
+
+# ---
+# get the machine-learning repo
+wget http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64/nvidia-machine-learning-repo-ubuntu1804_1.0.0-1_amd64.deb
+
+sudo apt install ./nvidia-machine-learning-repo-ubuntu1804_1.0.0-1_amd64.deb
+sudo apt-get update
```

Comment:
From https://docs.nvidia.com/deeplearning/cudnn/archives/cudnn_765/cudnn-install/index.html#cudnn-package-manager-installation-overview

> The Package Manager installation interfaces with your system's package manager. When using RPM or Deb, the downloaded package is a repository package, not the actual installation package. This repository package informs the package manager only where to find the actual installation packages, but will not install them.
> 
> If the actual installation packages are available in an online repository, they will be automatically downloaded in a later step. Otherwise, the repository package also installs a local repository containing the installation packages on the system.
> 
> Whether the repository is available online or installed locally, the installation procedure is identical and made of several steps. See below.